src/Conversion/InterventionImageFileConverter.php

```diff
@@ -0,0 +1,181 @@
+        }
+        return $result;
+    }
+
+    private function validateOptions(array $options): array
+    {
+        $problems = [];
+        foreach ($options as $key => $value) {
+            if ($key !== 'quality') {
+                $problems[] = "unexpected option '$key'";
+                continue;
+            }
+            if (!is_int($value)) {
+                $problems[] = "quality value must be an integer";
+            }
+        }
+        return $problems;
+    }
+
+    private function supportedByIntervention(string $format, InterventionBackend $backend): bool
+    {
+        $driver = $backend->getImageManager()->config['driver'] ?? null;
+        // If the driver is somehow not GD or Imagick, we have no way to know what it might support
+        if ($driver !== 'gd' && $driver !== 'imagick') {
+            return false;
+        }
+
+        // Return early for empty values - we obviously can't support that
+        if ($format === '') {
+            return false;
+        }
+
+        // GD and Imagick support different things.
+        // This follows the logic in intervention's AbstractEncoder::process() method
+        // and the various methods in the Encoder classes for GD and Imagick,
+        // excluding checking for strings that were obviously mimetypes
+        switch (strtolower($format)) {
+            case 'gif':
+                // always supported
+                return true;
+            case 'png':
+                // always supported
+                return true;
+            case 'jpg':
+            case 'jpeg':
+            case 'jfif':
+                // always supported
+                return true;
+            case 'tif':
+            case 'tiff':
+                if ($driver === 'gd') {
+                    false;
+                }
+                // always supported by imagick
+                return true;
+            case 'bmp':
+            case 'ms-bmp':
+            case 'x-bitmap':
+            case 'x-bmp':
+            case 'x-ms-bmp':
+            case 'x-win-bitmap':
+            case 'x-windows-bmp':
+            case 'x-xbitmap':
+                if ($driver === 'gd' && !function_exists('imagebmp')) {
+                    return false;
+                }
+                // always supported by imagick
+                return true;
+            case 'ico':
+                if ($driver === 'gd') {
+                    return false;
+                }
+                // always supported by imagick
+                return true;
+            case 'psd':
+                if ($driver === 'gd') {
+                    return false;
+                }
+                // always supported by imagick
+                return true;
+            case 'webp':
+                if ($driver === 'gd' && !function_exists('imagewebp')) {
+                    return false;
+                }
+                if ($driver === 'imagick' && !\Imagick::queryFormats('WEBP')) {
```

Comment:
```suggestion
                if ($driver === 'imagick' && !Imagick::queryFormats('WEBP')) {
```

Import Imagick up the top.  Remove the trailing slash for other examples in this file

Reply:
Done